tests/fp8/test_kernel.py

```diff
@@ -0,0 +1,34 @@
+import pytest
+import torch
+from nanotron.fp8.constants import DTypes
+from nanotron.fp8.kernel import _fp8_matmul_kernel
+from nanotron.fp8.tensor import FP8Tensor
+from torch import nn
+
+
+@pytest.mark.parametrize("hidden_size", [16, 64, 128, 256, 512, 1024])
+@pytest.mark.parametrize("transpose_a", [True, False])
+@pytest.mark.parametrize("transpose_b", [True, False])
+@pytest.mark.parametrize("use_split_accumulator", [True, False])
+def test_fp8_matmul_kernel(hidden_size, transpose_a, transpose_b, use_split_accumulator):
+    input = torch.randn(hidden_size, hidden_size, device="cuda")
+    linear = nn.Linear(hidden_size, hidden_size, bias=False, device="cuda")
+    weight = linear.weight.detach().clone()
+
+    input = input.T if transpose_a else input
+    weight = weight.T if transpose_b else weight
+
+    fp8_input = FP8Tensor(input.detach().clone(), DTypes.FP8E4M3)
+    fp8_weight = FP8Tensor(weight.detach().clone(), DTypes.FP8E4M3)
+
+    ref_output = torch.matmul(input, weight)
+
+    output = _fp8_matmul_kernel(
+        mat_a=fp8_input,
+        transpose_a=transpose_a,
+        mat_b=fp8_weight,
+        transpose_b=transpose_b,
+        use_split_accumulator=use_split_accumulator,
+    )
+
+    assert torch.allclose(output, ref_output, 0.1, 0.1)
```

Comment:
0.1 tolerance is a little too big to test anything? Let's try to calculate ref_output in a way to have 0 tolerance

Reply:
In the FP8-LM paper, they also use 0.1 as a tolerance, and they show that the convergence is acceptable for 125M, 7B, 13B, and 175B models!

Reply:
Oops. this is duplication test. We already have it here: https://github.com/huggingface/nanotron/blob/1702d21554d1c9fd2046e057d55356bac788e165/tests/fp8/test_linear.py#L36